File: src/main/java/com/hospital/management/patient/service/PatientService.java
package com.hospital.management.patient.service;

import com.hospital.management.patient.dto.PatientRequestDTO;
import com.hospital.management.patient.dto.PatientResponseDTO;

import java.util.List;

public interface PatientService {

    PatientResponseDTO savePatient(PatientRequestDTO dto);

    List<PatientResponseDTO> getAllPatients();

    PatientResponseDTO getPatientById(Long id);

    PatientResponseDTO updatePatient(Long id, PatientRequestDTO dto);

    void deletePatient(Long id);
}

File: src/services/patientService.js
import api from "./api";


export const getAllPatients = () => {
  return api.get("/api/patients");
};

export const getPatientById = (id) => {
  return api.get(`/api/patients/${id}`);
};

export const savePatient = (patient) => {
  return api.post("/api/patients", patient);
};

export const updatePatient = (id, patient) => {
  return api.put(`/api/patients/${id}`, patient);
};

export const deletePatient = (id) => {
  return api.delete(`/api/patients/${id}`);
};
export const getAllUsers = () => {
  return api.get("/api/users");
};
